src/store.ts

```diff
@@ -0,0 +1,113 @@
+import { types } from "mobx-state-tree";
+
+const validateMail = (str: string): boolean => {
+  let atIndex = str.indexOf("@");
+  let dotIndex = str.lastIndexOf(".");
+  return atIndex > 0 && dotIndex > atIndex;
+};
+
+const User = types.model({
+  name: types.string,
+  mail: types.string,
+  complaint: types.string,
+  selectedType: types.maybeNull(
+    types.enumeration(["Service related", "Product related"])
+  ),
+  createdAt: types.maybe(types.Date),
+});
+
+export const RootStore = types
+  .model({
+    name: types.maybe(types.string),
+    mail: types.maybe(types.string),
+    complaint: types.maybe(types.string),
+    selectedType: types.maybeNull(
+      types.enumeration(["Service related", "Product related"])
+    ),
+    createdAt: types.maybe(types.Date),
+    nameError: types.optional(types.boolean, false),
+    mailError: types.optional(types.boolean, false),
+    complaintError: types.optional(types.boolean, false),
+    users: types.array(User),
+  })
+  .actions((self) => ({
+    setName(value: string) {
+      self.name = value;
+      if (value.length > 16) this.setNameError(true);
+      else this.setNameError(false);
+    },
+    setMail(value: string) {
+      self.mail = value;
+      if (value && !validateMail(value)) this.setMailError(true);
+      else this.setMailError(false);
+    },
+
+    setComplaint(value: string) {
+      self.complaint = value;
+      if (value.length > 50) this.setComplaintError(true);
+      else this.setComplaintError(false);
+    },
+    setDate(value: Date) {
+      self.createdAt = value;
+    },
+    setNameError(value: boolean) {
+      self.nameError = value;
+    },
+    setMailError(value: boolean) {
+      self.mailError = value;
+    },
+    setComplaintError(value: boolean) {
+      self.complaintError = value;
+    },
+    setSelectedType(value: string | null) {
+      self.selectedType = value;
+    },
+    setUsers(value: {
+      name: string;
+      mail: string;
+      complaint: string;
+      selectedType: string | null;
+      createdAt: Date | undefined;
+    }) {
+      self.users.push(value);
+    },
+    handleSubmit() {
+      console.log("handleSubmit called");
+
+      if (self.name === "") this.setNameError(true);
+      if (self.mail === "") this.setMailError(true);
+      if (
+        self.complaint &&
+        (self.complaint === "" || self.complaint.length > 40)
+      )
+        this.setComplaintError(true);
+
+      if (
```

Comment:
Why is there a need to do the check multiple times? Please rethink this and come up with a better solution.